__tests__/app.test.js

```diff
@@ -352,3 +352,135 @@ describe("GET /api/users", () => {
+    return request(app)
+      .get("/api/articles?sort_by=title")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy("title", { descending: true });
+      });
+  });
+  test("200: Responds with an array of all articles sorted by topic", () => {
+    return request(app)
+      .get("/api/articles?sort_by=topic")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy("topic", { descending: true });
+      });
+  });
+  test("200: Responds with an array of all articles sorted by author", () => {
+    return request(app)
+      .get("/api/articles?sort_by=author")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy("author", { descending: true });
+      });
+  });
+  test("200: Responds with an array of all articles sorted by votes", () => {
+    return request(app)
+      .get("/api/articles?sort_by=votes")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy("votes", { descending: true });
+      });
+  });
+  test("400: Responds with relevant error to an invalid sort_by", () => {
+    return request(app)
+      .get("/api/articles?sort_by=body")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+  test("400: Responds with relevant error to a non-existent sort_by", () => {
+    return request(app)
+      .get("/api/articles?sort_by=not-a-column")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+});
+
+describe("GET /api/articles?order=asc", () => {
+  test("200: Responds with array of all articles sorted by created_at in ascending order", () => {
+    return request(app)
+      .get("/api/articles?order=asc")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy("created_at");
+      });
+  });
+  test("400: Responds with relevant error to an invalid order", () => {
+    return request(app)
+      .get("/api/articles?order=upside_down")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+});
+
+describe("GET /api/articles?sort_by=:column&&order=:order", () => {
+  test("200: Responds with an array of all articles sorted by article_id in ascending order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=article_id&&order=asc")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy("article_id");
+      });
+  });
```

Comment:
Nice, and same as the above only one of these combined tests would be fine, but having a test for error handling with both of these would be a nice to have as well